### DISCUSSION TITLE
[MTR05] 求助交作業問題

### DISCUSSION THREAD

**Discussion by hsujoey** [Lidemy/forum#59] [程式導師實驗計畫]
我的 repo : https://github.com/Lidemy/mentor-program-5th-hsujoey
<img width="580" alt="截圖 2021-04-19 19 41 22" src="https://user-images.githubusercontent.com/81914800/115231058-b4066500-a147-11eb-83e9-271d7316fc2e.png">

最初狀況說明：
1. 助教回覆說我作業只有傳到 hw5
2. 我上傳作業時，push 去 master了。

後續狀況說明：
1. 今天嘗試把 origin master 恢復原狀，想法是：在我的 iTerm上切到 master 這個 branch ，回到最一開始沒任何 commit 的版本，再 push 回去。 不過整個過程有很多狀況，最後沒成功，還動了很多東西。（中間過程有改有刪，如果需要我再補充）

2. 總之，放棄把 origin master 恢復原狀後，已經亂糟糟到自己看不懂了😂
於是我把電腦上的 mentor-5th 刪掉，重新 clone 下來，開 branch week1  並把 hw1~hw5 編輯好 ， commit hw-all ，要 push 回去，然後碰到圖片中的狀況。

問題：想請問接下來處理～ 謝謝🙏

**Comment by qoosuperman** [comment]
小弟我嘗試回答看看
1. 可以用 `git reflog` 看看你之前 head 移動過程，看紀錄找不找得回來
2. 如果你只是想把本地還原成跟某個 remote branch 一樣的話可以嘗試以下步驟： 把本地端資料 reset 到跟 remote 那個 branch 一樣 /  checkout 到那個 branch / 再試試看是否能正常 pull，指令是什麼的話再麻煩找資料或自己想想看會比較熟悉~

**Comment by aszx87410** [comment]
應該是因為你遠端的 branch 沒有刪除所以衝突了，試著把遠端的相同名稱分支刪掉：https://github.com/Lidemy/mentor-program-5th-hsujoey/branches
